src/cli_plugin/install/downloaders/http.js

```diff
@@ -2,11 +2,47 @@ import Wreck from 'wreck';
 import Progress from '../progress';
 import { fromNode as fn } from 'bluebird';
 import { createWriteStream } from 'fs';
+import HttpProxyAgent from 'http-proxy-agent';
+import URL from 'url';
 
-function sendRequest({ sourceUrl, timeout }) {
+function getProxyAgent(sourceUrl, logger) {
+  const httpProxy = process.env.http_proxy || process.env.HTTP_PROXY;
+  // we have a proxy detected, lets use it
+  if (httpProxy) {
+    logger.log(`Picked up proxy ${httpProxy} from http_proxy environment variable.`);
+    // get the hostname of the sourceUrl
+    const hostname = URL.parse(sourceUrl).hostname;
+    const noProxy = process.env.no_proxy || process.env.NO_PROXY || '';
+    const excludedHosts = noProxy.split(',');
+
+    // proxy if the hostname is not in noProxy
+    const shouldProxy = (excludedHosts.indexOf(hostname) === -1);
```

Comment:
Btw, saw https://github.com/Rob--W/proxy-from-env. If it makes our lives easier we could just fork it and bring it into the codebase and make the necessary changes. It also has a good test suite we can use as a starting point: https://github.com/Rob--W/proxy-from-env/blob/master/test.js